File: Part_2/Matlab/detectWalking/testDetectWalkingFunction.m
%% Test your code by running the following
close all;
clear all;
testDetectWalking();

function testDetectWalking()
    files = dir('dataset');
    files = files(3:end,1);

    for i = 1:size(files,1)
        filename = files(i).name;
        data_table = readtable(strcat('dataset/',filename));
        
        detectWalkingPredictions = detectWalking(data_table);
        
        figure
        plot(data_table.movement, 'b')
        hold on;
        plot(detectWalkingPredictions,'r')
    end
end

File: Part_2/Python/detectWalking/testDetectWalkingFunction.py
#!/usr/bin/env python
import detectWalking as dw
import matplotlib.pyplot as mplt
import os, sys
currentdir = os.path.dirname(os.path.realpath(__file__))
parentdir = os.path.dirname(currentdir)
sys.path.append(parentdir)
import helperFunctions as hf

# Test your code by running the following

def testDetectWalking():
	files = os.listdir('dataset')

	i = 0
	for filename in files:
		data_array = hf.load_csv_file('dataset/' + filename)
		
		detectWalkingPredictions = dw.detectWalking(data_array)
		
		mplt.figure(i+1)
		mplt.plot(data_array[:,hf.variable_name_to_column_index('movement')], 'b')
		mplt.plot(detectWalkingPredictions,'r')
		mplt.draw()

		i += 1

	mplt.show()

if __name__ == "__main__":
	testDetectWalking()